internal/vici/listConns.go

```diff
@@ -24,6 +23,18 @@ func (c *ClientConn) ListConns(ike string) (map[string]IKEConf, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error registering list-conn event: %w", err)
 	}
+
+	defer func() {
+		unregisterErr := c.UnregisterEvent("list-conn")
+		if unregisterErr != nil {
+			if err == nil {
+				err = unregisterErr
+				return
```

Comment:
Is this correct? Can we add some tests to show the correct behavior?

Reply:
Yes it is. We are using the dreaded named return values so line 31 assigns the return value of the function. We then return the deferred function to do nothing else.

We cannot test this as is as the client is coupled directly to the vici daemon :( An interface would help here, but I haven't come around to it...

Reply:
Lazy developers excuse 🙈